onyo/lib/git.py

```diff
@@ -354,65 +329,63 @@ def get_hexsha(self,
         ----------
         commitish
             Any identifier that refers to a commit (defaults to "HEAD").
-
-        Returns
-        -------
-        str
-            the commit message including the subject line.
         """
-        return self._git(['log', commitish or 'HEAD', '-n1', '--pretty=%B'])
 
-    def check_ignore(self, ignore: Path, paths: list[Path]) -> list[Path]:
-        r"""Get the `paths` that are matched by patterns defined in `ignore`.
+        return self._git(['log', commitish or 'HEAD', '-n1', '--pretty=%B'])
 
-        This is utilizing ``git-check-ignore`` to evaluate `paths` against
-        a file `ignore`, that defines exclude patterns the gitignore-way.
+    def check_ignore(self,
+                     ignore: Path,
+                     paths: list[Path]) -> list[Path]:
+        r"""Get the subset of ``paths`` that are matched by patterns defined in
+        the ``ignore`` file.
 
         Parameters
         ----------
         ignore
-          Path to a file containing exclude patterns to evaluate.
+            Path to a file containing exclude patterns (in the style of ``.gitignore``).
         paths
-          Paths to check against the patterns in `ignore`.
-
-        Returns
-        -------
-        list of Path
-          Paths in `paths` that are excluded by the patterns in `ignore`.
+            Paths to check
         """
+
         try:
             output = self._git(['-c', f'core.excludesFile={str(ignore)}', 'check-ignore', '--no-index', '--verbose'] +
                                [str(p) for p in paths])
         except subprocess.CalledProcessError as e:
             if e.returncode == 1:
-                # None of `paths` was ignored. That's fine.
+                # None of `paths` were ignored. That's ok.
                 return []
-            raise  # reraise on unexpected error
+            raise
+
         excluded = []
         for line in output.splitlines():
             parts = line.split('\t')
             src_file = Path(parts[0].split(':')[0])
             if src_file == ignore:
                 excluded.append(Path(parts[1]))
+
         return excluded
 
-    # TODO: git check-ignore --no-index --stdin (or via command call)  ->  lazy, check GitRepo.files once. (Same invalidation)
-    #       -> OnyoRepo would use it to maintain a ignore list from a (top-level .onyoignore)? .onyo/ignore ? Both?
+    def _parse_log_output(self,
+                          lines: list[str]) -> dict:
+        """Produce a commit dict (of one commit) from the output of ``git log``.
 
-    def _parse_log_output(self, lines: list[str]) -> dict:
-        """Generate a dict from the output of ``git log`` for one commit.
+        A helper for :py:func:`history`.
 
-        Internal helper that converts a list of git-log-output lines of
-        a single commit to a dictionary.
+        Parameters
+        ----------
+        lines
+            List of text from ``git log`` of a single commit.
```

Comment:
I think I know what this explanation means, but it is the least clear parameter explanation I read so far, maybe it can be descripted more clearly.

Is each element in the list one line of the `git log` output, including some for empty lines, some for the operation record and so on?

Reply:
I have tweaked the wording here a bit, but it's not worth much effort as it is a private function that is only ever used in the context of someone who is actively altering `GitRepo.history()`.

Indeed, your understanding is correct. It is a list of lines of text for one commit that is output from `git log`.